aztk/client/base/base_operations.py

```diff
@@ -0,0 +1,218 @@
+from aztk import models
+from aztk.internal import cluster_data
+from aztk.utils import ssh as ssh_lib
+
+from .helpers import (create_user_on_cluster, create_user_on_node, delete_user_on_cluster, delete_user_on_node,
+                      generate_user_on_cluster, generate_user_on_node, get_application_log, get_remote_login_settings,
+                      node_run, run, ssh_into_node)
+
+
+class BaseOperations:
+    """Base operations that all other operations inherit from
+
+    Attributes:
+        batch_client (:obj:`azure.batch.batch_service_client.BatchServiceClient`): Client used to interact with the
+            Azure Batch service.
+        blob_client (:obj:`azure.storage.blob.BlockBlobService`):  Client used to interact with the Azure Storage
+            Blob service.
+        secrets_configuration (:obj:`aztk.models.SecretsConfiguration`): Model that holds AZTK secrets used to authenticate
+            with Azure and the clusters.
+    """
+
+    def __init__(self, context):
+        self.batch_client = context['batch_client']
+        self.blob_client = context['blob_client']
+        self.secrets_configuration = context['secrets_configuration']
+
+    def get_cluster_config(self, id: str) -> models.ClusterConfiguration:
+        """Open an ssh tunnel to a node
+
+        Args:
+            id (:obj:`str`): the id of the cluster the node is in
+            node_id (:obj:`str`): the id of the node to open the ssh tunnel to
+            username (:obj:`str`): the username to authenticate the ssh session
+            ssh_key (:obj:`str`, optional): ssh public key to create the user with, must use ssh_key
+                or password. Defaults to None.
+            password (:obj:`str`, optional): password for the user, must use ssh_key or password. Defaults to None.
+            port_forward_list (:obj:`List[PortForwardingSpecification`, optional): list of PortForwardingSpecifications.
+                The defined ports will be forwarded to the client.
+            internal (:obj:`bool`, optional): if True, this will connect to the node using its internal IP.
+                Only use this if running within the same VNET as the cluster. Defaults to False.
+        Returns:
+            ClusterConfiguration: Object representing the cluster's configuration
```

Comment:
shouldn't that also be a type?

Reply:
what do you mean?

Reply:
`:obj`ClusterConfiguration``

Reply:
these actually weren't even being published, but I changed that so now they are fixed and are being published